src/main/java/seedu/address/model/event/StartTime.java

```diff
@@ -14,46 +14,50 @@
  */
 public class StartTime implements Comparable<StartTime> {
 
-    public static final String MESSAGE_FORMAT_CONSTRAINTS = "Start time must be in format: hh:mm.";
+    public static final String MESSAGE_FORMAT_CONSTRAINTS = "Start time must be in format: hh:mm,"
+            + " hh and mm must be zero padded number of length 2, eg. 02:30";
     public static final String MESSAGE_VALUE_CONSTRAINTS = "%s exceeds the range of valid time value.";
-    private static final DecimalFormat df = new DecimalFormat("00");
+    private static final DecimalFormat TIME_FORMATTER = new DecimalFormat("00");
+    private static final String VALIDATION_REGEX = "\\d{2}:\\d{2}";
     public final LocalTime time;
 
-
     /**
      * Constructs a {@code StartTime}.
      * @param startTime A valid start time.
      */
     public StartTime(String startTime) {
         requireNonNull(startTime);
         assert !startTime.isBlank();
-        checkArgument(isValidStartTime(startTime) != 0, MESSAGE_FORMAT_CONSTRAINTS);
-        checkArgument(isValidStartTime(startTime) != -1, String.format(MESSAGE_VALUE_CONSTRAINTS, startTime));
+        checkArgument(isValidStartTime(startTime), MESSAGE_FORMAT_CONSTRAINTS);
+        checkArgument(isValidStartTimeValue(startTime), String.format(MESSAGE_VALUE_CONSTRAINTS, startTime));
         String [] parsedTime = startTime.split(":", 2);
-        this.time = LocalTime.of(Integer.parseInt(parsedTime[0]), Integer.parseInt(parsedTime[1]), 0);
+        this.time = LocalTime.of(Integer.parseInt(parsedTime[0]), Integer.parseInt(parsedTime[1]));
     }
 
     /**
-     * Checks if a given string is a valid StartTime input.
-     * @return int value, 0 represent invalid time format, -1 represents invalid time value, 1 represents valid time.
+     * Checks if a given string follows the valid StartTime input format.
+     * @return boolean value.
      */
-    public static int isValidStartTime(String test) {
+    public static boolean isValidStartTime(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * Checks if a given string can be parsed to a valid StartTime value.
+     * @param test string input, it is guaranteed that the format check for test is done before this function.
```

Comment:
I don't think it's a good idea to guarantee that the format check is done before this function, because a method can't guarantee who calls it or when it's called.

```suggestion
     * @param test string input, the format check for test should be done before calling this function.
```